src/swsssdk/configdb.py

```diff
@@ -198,14 +200,44 @@ def mod_entry(self, table, key, data):
         if data == None:
             client.delete(_hash)
         else:
-            client.hmset(_hash, self.__typed_to_raw(data))
+            # We allow deletion of field by setting them to None, __typed_to_raw()
+            # will fill keys to be deleted in dkeys arg.
+            dkeys = dict()
+            raw = self.__typed_to_raw(data, dkeys)
+            if raw and len(raw):
+                client.hmset(_hash, raw)
+            # check if any dkeys(i.e. fields) to be deleted
+            if len(dkeys):
+                # get the original first, nothing to do if original is None
+                original = self.get_entry(table, key)
+                if original:
+                    self.__delete_fields(client, _hash, dkeys, original)
```

Comment:
the function interfaces is confusing, either we should have "table, _hash, key" etc as input, or we can simply do an inline functions here, not having a separate function to save passing parameters around.

Reply:
let me try with inline functions and reusing same args.

Reply:
Done, no extra functions.